app/views/reports/elements/_my_module_element.html.erb

```diff
@@ -74,13 +74,7 @@
     <% end %>
 
     <% order_results_for_report(my_module.results, @settings.dig('task', 'result_order')).each do |result| %>
-      <% if result.is_asset && @settings.dig('task', 'file_results') %>
         <%= render partial: 'reports/elements/my_module_result_asset_element', locals: { result: result, report: report, export_all: export_all } %>
-      <% elsif result.is_table && @settings.dig('task', 'table_results') %>
-        <%= render partial: 'reports/elements/my_module_result_table_element', locals: { result: result, export_all: export_all } %>
-      <% elsif result.is_text && @settings.dig('task', 'text_results') %>
-        <%= render partial: 'reports/elements/my_module_result_text_element', locals: { result: result, export_all: export_all } %>
-      <% end %>
```

Comment:
This needs to be re-implemented using the result_orderable_element association. Loop through each element and render it based on the orderable type.